### PR TITLE
Encode as bytes1 instead of uint8

### PR DESCRIPTION
**Pull request by expede**
**Summary**
Switch from `uint8` to `byte`, and use `hex` in examples. This brings us in line with the most recent Ethereum Status Codes.

Also added a few newlines to keep line length under 100 characters / readability. Still figuring out style in Solidity, but I think this may be easier to read.

### CODE REVIEW

File: contracts/SimpleAuthorization.sol [hunk 1]
```diff
@@ -6,25 +6,35 @@ import "./TokenValidator.sol";
...
-  }
+    function check(
+        address /* token */,
+        address _address
+    ) public /* view */ returns (byte resultCode) {
+        if (auths[_address]) {
+            return hex"11";
+        } else {
+            return hex"10";
+        }
```
Comment (expede) [inline contracts/SimpleAuthorization.sol]: Not totally sure _why_ it didn't like this as a ternary operator. I've run into this a few times.

File: test/spec/simple-authorization.spec.js [hunk 2]
```diff
@@ -16,34 +16,34 @@ contract('SimpleAuthorization', (accounts) => { // eslint-disable-line no-undef
...
 
     await simpleAuthorization.setAuthorized(receiver, true);
     let authorizedAfter = await simpleAuthorization.check4.call(0x0, sender, receiver, 0);
-    assert.equal(authorizedAfter, 0, 'should still be unauthorized');
+    assert.equal(authorizedAfter, '0x10', 'should still be unauthorized');
 
     await simpleAuthorization.setAuthorized(sender, true);
     authorizedAfter = await simpleAuthorization.check4.call(0x0, sender, receiver, 0);
-    assert.equal(authorizedAfter, 1, 'should become authorized');
+    assert.equal(authorizedAfter, '0x11', 'should become authorized');
```
Comment (expede) [inline test/spec/simple-authorization.spec.js]: Interesting how the JS client interprets hex data. Solidity treats it as raw bytes, Chai treats it as a string 🤷🏻‍♀️

File: contracts/SimpleAuthorization.sol [hunk 3]
```diff
@@ -6,25 +6,35 @@ import "./TokenValidator.sol";
...
+    function check(
+        address /* _token */,
+        address _from,
+        address _to,
+        uint256 /* _amount */
+    ) public /* view */ returns (byte resultCode) {
+        if (auths[_from] && auths[_to]) {
+            return hex"11";
+        } else {
+            return hex"10";
```
[the remark this answers is not in the document]
Comment (carchrae) [reply contracts/SimpleAuthorization.sol]: seems constants can be defined, but not entirely clear if this fits here

http://solidity.readthedocs.io/en/v0.4.21/contracts.html#constant-state-variables

[the remark this answers is not in the document]
Comment (expede) [reply contracts/SimpleAuthorization.sol]: Indeed. The magic codes live in the [`EthereumStatusCodes`](https://github.com/Finhaven/EthereumStatusCodes) repo. [These are those enums](https://github.com/Finhaven/EthereumStatusCodes/pull/5/files#diff-4ae6eb8ca2a33e72c0b7cc8669e31f07R3) that I was talking about yesterday. It's a WIP right now 😬 I'm getting it in shape for EIPing.

File: contracts/SimpleAuthorization.sol (hunk 1 above)
Comment (expede) [reply contracts/SimpleAuthorization.sol]: I tried the following variations:

```solidity
(auths[_address]) ? return hex"11" :  return hex"10";
(auths[_address]) ? (return hex"11") :  (return hex"10");
return (auths[_address]) ? hex"11" :  hex"10";
return ((auths[_address]) ? hex"11" :  hex"10");
return ((auths[_address]) ? (hex"11") :  (hex"10"));
```

None of these compiled. It complained that I wasn't returning a `byte`. Switch to `if/else` syntax works 🤷🏻‍♀️

File: test/spec/simple-authorization.spec.js (hunk 2 above)
Comment (expede) [reply test/spec/simple-authorization.spec.js]: Good point; I'll add a helper JS lib 👌🏻

File: contracts/SimpleAuthorization.sol (hunk 1 above)
Comment (carchrae) [reply contracts/SimpleAuthorization.sol]: weird - maybe a solidity shortcoming.  or maybe disdain for the elvis operator.  reading this: http://solidity.readthedocs.io/en/v0.4.21/control-structures.html#output-parameters

i suspect that `result = auths[_address] ? 1 : 0;` might actually work.  or even, simply `auths[_address] ? 1 : 0;` - ie, skip the return as it is the last line

Comment (carchrae) [reply contracts/SimpleAuthorization.sol]: i'm not really a fan of the implict return.  i'm know i'm not a rubyist.

Comment (expede) [reply contracts/SimpleAuthorization.sol]: > i suspect that result = auths[_address] ? 1 : 0; might actually work

I've had this work with assignment, yeah

> simply auths[_address] ? 1 : 0; - ie, skip the return as it is the last line

I'm pretty sure that you need an explicit `return`. I see it in all the one-liners, for instance.

Comment (carchrae) [reply contracts/SimpleAuthorization.sol]: btw, none of this is a big deal, just trying to understand the quirks of solidity.

**Comment by naumenkogs** [comment]
I'm wondering about the *view* commented here.
https://github.com/Finhaven/ValidatedToken/blob/f2a8d3970735ff49af14338d88881f98b42881cf/contracts/SimpleAuthorization.sol#L14


I would go through modifiers like *pure*/*external*/*private* and check that those are used properly. Other than that it looks fine to me.

**Comment by expede** [comment]
@naumenkogs totally agree. The visibility levels are super confusing. The latest version of Truffle (or maybe just solc?) is trying to encourage using `external` and `internal` more, which makes sense. Just our deps aren't updated yet.

AFAICT works like this:

## `external`
Treat function as totally external. Need to use `this.` to access even from the same contract. Makes it very clear when using interfaces.

## `public`
External callers may access this method. Internal calls don't require `this.`

## `internal`
Only callable within the contract. Accessible by inherited contracts.

## `private`
Only callable within the contract. _Not accessible by inheritable contracts._

File: contracts/ReferenceToken.sol [hunk 4]
```diff
@@ -117,19 +129,29 @@ contract ReferenceToken is Ownable, ERC20, ValidatedToken {
...
+        mBalances[_to] = mBalances[_to].add(_amount);
 
-          mBalances[_from] = mBalances[_from].sub(_amount);
-          mBalances[_to] = mBalances[_to].add(_amount);
-          Transfer(_from, _to, _amount);
-      }
+        emit Transfer(_from, _to, _amount);
+    }
 
+    function canTransfer(
```
Comment (naumenkogs) [inline contracts/ReferenceToken.sol]: why don't we add requireMultiple to this method?

I don't think you'll ever be able to transfer a fraction of a token :)

**Comment by naumenkogs** [comment]
I'm also not getting why hex"11" is a success :)

I understand it's temporary, but still

**Comment by expede** [comment]
@naumenkogs tests fixed 🙌🏻

**Pull request merged**